smac/scenario/scenario.py

```diff
@@ -250,12 +255,24 @@ def extract_instance_specific(instance_list):
                     self.feature_fn)[1]
 
         if self.feature_dict:
-            self.n_features = len(
-                self.feature_dict[list(self.feature_dict.keys())[0]])
             self.feature_array = []
             for inst_ in self.train_insts:
                 self.feature_array.append(self.feature_dict[inst_])
             self.feature_array = numpy.array(self.feature_array)
+            
+            # reduce dimensionality of features of larger than PCA_DIM
+            if self.feature_array.shape[1] > self.PCA_DIM:
+                X = self.feature_array
+                # scale features
```

Comment:
From my understanding of [the code](https://github.com/scikit-learn/scikit-learn/blob/a5ab948/sklearn/decomposition/pca.py#L293), PCA does the scaling itself. Otherwise, I think you should use the MinMaxScaler form scikit-learn.